### PR TITLE
feat: remove 'databases' from breadcrumb and replace with pulldown to switch schemas

### PR DESCRIPTION
**Pull request by mswertz**
so tenants really feel in their own database

To test: open random schema and see 'databases' is not in bread crumb.
Instead there is a pulldown to switch to other schemas

See https://preview-pr-emx2-834.dev.molgenis.org/apps/central/#/
(add a few schema's, and then open one and you will see it).

image

### CODE REVIEW

**Comment by connoratrug** [comment]
add the admin flag as prop ?, maybe non admins would like to have it in case of managing a list of staging areas. Or leave it like this and make sure the menu logo links to home ( databases in case of admin )

**Comment by mswertz** [comment]
[Remove 'databases' from breadcrumb](https://trello.com/c/hemKD8Ky/357-remove-databases-from-breadcrumb)

**Review by connoratrug — CHANGES_REQUESTED** [review]
tests a failing and preview does not work

**Comment by mswertz** [comment]
> tests a failing and preview does not work

seems to all work fine now?

**Comment by erikzwart** [comment]
Pulldown to select database works fine:
PR-834_breadcrumb

Is it possible to indent the databases or something like:
databases:
- pet store
- catalogue

**Comment by mswertz** [comment]
got approval from Erik.

File: apps/styleguide/src/layout/Molgenis.vue [hunk 1]
```diff
@@ -78,13 +79,26 @@ export default {
...
+      if (this.session && this.session.schemas) {
+        result["databases:"] = "/";
+        this.session.schemas.forEach((s) => {
+          result["- " + s] = "../../" + s; // all paths are of form /:schema/:app
+        });
+      }
+      if (result.length > 0) {
+        return result;
+      }
+      return result;
```
Comment (connoratrug) [inline apps/styleguide/src/layout/Molgenis.vue:93]: the if does not do anything

File: backend/molgenis-emx2-graphql/src/main/java/org/molgenis/emx2/graphql/GraphqlSessionFieldFactory.java [hunk 2]
```diff
@@ -117,6 +122,7 @@ public GraphQLFieldDefinition userQueryField(Database database, Schema schema) {
               if (schema != null) {
                 result.put(ROLES, schema.getInheritedRolesForActiveUser());
               }
+              result.put(SCHEMAS, database.getSchemaNames());
```
Comment (connoratrug) [inline backend/molgenis-emx2-graphql/src/main/java/org/molgenis/emx2/graphql/GraphqlSessionFieldFactory.java:125]: all schemas or all readable by user ?

Comment (mswertz) [reply backend/molgenis-emx2-graphql/src/main/java/org/molgenis/emx2/graphql/GraphqlSessionFieldFactory.java:125]: this is automatically filtered to only show schemas available to user

File: apps/styleguide/src/layout/Molgenis.vue [hunk 3]
```diff
@@ -78,13 +79,26 @@ export default {
     };
   },
   computed: {
+    schemaUrlsForCrumbs() {
+      var result = {};
+      if (this.session && this.session.schemas) {
+        result["databases:"] = "/";
+        this.session.schemas.forEach((s) => {
+          result["- " + s] = "../../" + s; // all paths are of form /:schema/:app
```
[the remark this answers is not in the document]
Comment (mswertz) [reply apps/styleguide/src/layout/Molgenis.vue:87]: at request of Erik and I agree: then in the user interface it is easier to distinguish the first link (which is back to the database listing) from the schema links. I considers bullets and stuff but then the pull request became unnecessary complicated

[the remark this answers is not in the document]
Comment (mswertz) [reply apps/styleguide/src/layout/Molgenis.vue]: removed it

**Pull request merged**